orttraining/orttraining/python/training/ortmodule/_graph_transition_manager.py

```diff
@@ -0,0 +1,783 @@
+                if PrimitiveType.is_primitive_type(data) and constant_as_tensor:
+                    data = PrimitiveType.get_tensor(data, device)
+                self._buffer_for_ort_runs[name] = data
+            else:
+                raise wrap_exception(
+                    ORTModuleONNXModelException,
+                    RuntimeError(f"Input is present in ONNX graph but not provided: {name}."),
+                )
+
+        return self._buffer_for_ort_runs
+
+    def restore_outputs(self, ort_flatten_outputs: list[torch.Tensor]):
+        """Restores the outputs from the ORT forward run, back to the original data structure"""
+
+        try:
+            return unflatten_data_using_schema(ort_flatten_outputs, self.module_forward_output_schema)
+        except TypeError as e:
+            raise wrap_exception(
+                ORTModuleIOError,
+                TypeError(f"ORTModule fails to unflatten user output: {e}"),
+            ) from None
+
+
+class GraphTransitionManager:
+    """Manage the graph transition from 1). PyTorch to ONNX export and 2). ONNX to ONNX post-export processing."""
+
+    def __init__(
+        self,
+        flatten_module: torch.nn.Module,
+        export_mode: int,
+        debug_options: DebugOptions,
+        runtime_options: _RuntimeOptions,
+        time_tracker: TimeTracker,
+        logger: logging.Logger,
+    ):
+        self._device = _utils._get_device_from_module(flatten_module)
+        self._export_mode = export_mode
+
+        self._debug_options = debug_options
+        self._runtime_options = runtime_options
+
+        self._export_extra_kwargs = {}
+
+        self._logger = logger
+
+        # Tracker for ORTModule model export.
+        self._time_tracker = time_tracker
+
+        # A signal to indicate if the original model has changed and need a re-export.
+        self._original_model_has_changed = False
+
+        self._flatten_module = flatten_module
+
+        # Forward function input parameters of the original module.
+        self._module_forward_func_parameters: list[inspect.Parameter] = list(
+            inspect.signature(self._flatten_module._original_module.forward).parameters.values()
+        )
+        # TODO: remove after PyTorch ONNX exporter supports VAR_KEYWORD parameters.
+        for input_parameter in self._module_forward_func_parameters:
+            if input_parameter.kind == inspect.Parameter.VAR_KEYWORD:
+                logger.info("The model's forward method has **kwargs parameter which has EXPERIMENTAL support!")
+
+        # Model info collected from the original module's forward function signature and args/kwargs, used for ONNX export.
+        self._model_info_for_export: _io.ModelInfoForExport | None = None
+        self._exported_model_info: ExportedModelInfo | None = None
+
+        # Model info after export and post export processing.
+        self._post_export_processed_model_info = None
+
+    def use_cache_or_reconstruct_post_processed_model(
```

Comment:
Using the cache or reconstructing the graph can be internal to this function and does not need to feature in the function name.
```suggestion
    def get_post_processed_model(
```